vizro-core/src/vizro/models/_components/table.py

```diff
@@ -51,12 +53,35 @@ def __getitem__(self, arg_name: str):
             return self.type
         return self.figure[arg_name]
 
+    def _build_underlying_table_object(self):
+        data = data_manager._get_component_data(self.id)  # type: ignore
+        kwargs = self.figure._arguments.copy()
+        kwargs.pop("data_frame", None)
+        return self.figure._function(data_frame=data, **kwargs)
+
+    @_log_call
+    def pre_build(self):
+        if self.actions:
+            # The underlying table object is pre-built, so we can fetch its ID.
+            underlying_table_object = self._build_underlying_table_object()
```

Comment:
Do we need to build the actual table for this? Could we not just build an empty table? Then we can get rid of this whole arguments fetching stuff?

Reply:
I refactored and avoided to create a temporary datatable with the original data_frame. Does it look good now?